src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -188,6 +187,11 @@ private static void SetMSBuildMetadata(IObjectWriter writer, PackageSpec package
                 SetValue(writer, "packagesConfigPath", pcMsbuildMetadata.PackagesConfigPath);
             }
 
+            if (packageSpec.RestoreMetadata.SdkAnalysisLevel is not null)
+            {
+                SetValue(writer, "SdkAnalysisLevel", packageSpec.RestoreMetadata.SdkAnalysisLevel.ToString());
```

Comment:
packageSpec.RestoreMetadata.SdkAnalysisLevel is a NugetVersion type, and it could be null. We are doing the null check here to prevent null reference when we do .ToString() before calling SetValue

Reply:
Oh yeah great point. I missed that when reviewing. I assumed it was a string.